### PR TITLE
POL-173 Update Azure policies for credential service 2

### PR DESCRIPTION
**Pull request by KTebrewal**
### Description

Update listed Azure policies to support the credential service. The changes include-
-Replace auth declaration with the credential declaration.
-Remove long_description field and use new info hash for Version, providers and service.
-Update readme to replace credential instructions with instructions to use credential service.
-Update CWF definitions as needed.

### Issues Resolved

Changed the authentication to credential services.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running template Link

**Azure AHUB Utilization with Manual Entry**
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e298c1bcbdfbd000104c1dd
image

**Azure Hybrid Use Benefit for Windows Server**
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e297f234ec7b300014e6b79
image

**Azure Regulatory Compliance**
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e3bc905cbdfbd00012f8ea5
image

**Azure VMs Not Using Managed Disks**
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e2fdba006101000014d5f1a

### CODE REVIEW

File: compliance/azure/ahub_manual/README.md [hunk 1]
```diff
@@ -1,57 +1,38 @@
...
+- *Allowed AHUB licenses* - Number of AHUB licenses that are allowed to be run on Azure
+- *Exclusion Tag Key* - Azure VMs instance tag to ignore instance that are with AHUB enabled. Only supply the tag key. The policy assumes that the tag value is irrelevant.
 
-1. Follow steps to [Create an Azure Active Directory Application](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#create-an-azure-active-directory-application)
-1. Grant the Azure AD Application access to the necessary subscription(s)
-1. [Retrieve the Application ID & Authentication Key](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-application-id-and-authentication-key)
-1. Create RightScale Credentials with values that match the Application ID (Credential name: `AZURE_APPLICATION_ID`) & Authentication Key (Credential name: `AZURE_APPLICATION_KEY`)
-1. [Retrieve your Tenant ID](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-tenant-id)
+## Prerequesites
```
Comment (cdwilhelm) [inline compliance/azure/ahub_manual/README.md]: The prerequisite section is missing the provider tag and cloud configuration sections.  See completed policies for an example

File: compliance/azure/ahub_manual/README.md [hunk 2]
```diff
@@ -1,57 +1,38 @@
...
 
-This policy has the following input parameters required when launching the policy.
+The following Azure permissions must be allowed for the policy to run.
 
-- *Azure AD Tenant ID* - the Azure AD Tenant ID used for the Azure API Authentication
-- *Azure Subscription ID* - the Azure Subscription ID used for the Azure API Authentication
-- *Email addresses to notify* - Email addresses of the recipients you wish to notify when new incidents are created
-- *Allowed AHUB licenses* - Number of AHUB licenses that are allowed to be run on Azure
-- *Exclusion Tag Key* - Azure VMs instance tag to ignore instance that are with AHUB enabled. Only supply the tag key. The policy assumes that the tag value is irrelevant.
+- Microsoft.Compute/virtualMachines/read
```
Comment (cdwilhelm) [inline compliance/azure/ahub_manual/README.md]: Are there other permissions needed for AHUB?

File: compliance/azure/compliance_score/README.md [hunk 3]
```diff
@@ -1,34 +1,19 @@
...
-### Input Parameters
+## Input Parameters
 
 This policy has the following input parameters required when launching the policy.
 
-- *Azure AD Tenant ID* - the Azure AD Tenant ID used for the Azure API Authentication
-- *Azure Subscription ID* - the Azure Subscription ID used for the Azure API Authentication
 - *Email addresses of the recipients you wish to notify* - A list of email addresses to notify
 
+## Pre-requisites
```
Comment (cdwilhelm) [inline compliance/azure/compliance_score/README.md]: The prerequisite section is missing the provider tag and cloud configuration sections.  See completed policies for an example.

File: compliance/azure/compliance_score/azure_regulatory_compliance_report.pt [hunk 4]
```diff
@@ -97,6 +123,10 @@ script "js_calculate_score", type: "javascript" do
 EOS
 end
 
+##########
+# Policy
+##########
+
 policy "policy_azure_exec_reg_compliance" do
   validate $ds_format_report do
     summary_template "Regulatory Compliance Report"
```
Comment (cdwilhelm) [inline compliance/azure/compliance_score/azure_regulatory_compliance_report.pt]: Please add and sort by Subscription Name

File: cost/azure/hybrid_use_benefit/README.md [hunk 5]
```diff
@@ -1,49 +1,35 @@
...
 
 This policy has the following input parameters required when launching the policy.
 
-- *Azure AD Tenant ID* - the Azure AD Tenant ID used for the Azure API Authentication
-- *Azure Subscription ID* - the Azure Subscription ID used for the Azure API Authentication
 - *Exclusion Tag Key* - an Azure-native instance tag to ignore instances that you don't want AHUB applied to. Only supply the tag key
 - *Email addresses of the recipients you wish to notify* - A list of email addresses to notify
 
-### Supported Clouds
+## Prerequisites
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit/README.md]: The prerequisite section is missing the provider tag and cloud configuration sections.  See completed policies for an example.  Are there additional permissions needed for Ahub?

File: cost/azure/hybrid_use_benefit/README.md [hunk 6]
```diff
@@ -1,49 +1,35 @@
...
-
-1. Follow steps to [Create an Azure Active Directory Application] (https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#create-an-azure-active-directory-application)
-1. Grant the Azure AD Application access to the necessary subscription(s)
-1. [Retrieve the Application ID & Authentication Key] (https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-application-id-and-authentication-key)
-1. Create RightScale Credentials with values that match the Application ID (Credential name: `AZURE_APPLICATION_ID`) & Authentication Key (Credential name: `AZURE_APPLICATION_KEY`)
-1. [Retrieve your Tenant ID] (https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-tenant-id)
-
-### Functional Details
-
-_ The policy identifies all Windows server instances that are not currently using [Azure Hybrid Use Benefit] (https://azure.microsoft.com/en-us/pricing/hybrid-benefit/). It raises an incident for all applicable VMs not currently using AHUB, which once approved, will enable AHUB on all identified instances.
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit/README.md]: The text doesn't have a link. check the markdown.

File: cost/azure/hybrid_use_benefit/azure_hybrid_use_benefit.pt [hunk 7]
```diff
@@ -130,18 +165,24 @@ script "js_filter_resources", type: "javascript" do
         }
       }
     })
+    result = _.sortBy(result, 'location');
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit/azure_hybrid_use_benefit.pt]: should be sorting by subscription name first

File: operational/azure/vms_without_managed_disks/README.md [hunk 8]
```diff
@@ -11,30 +11,19 @@ When a VM that is using unmanaged disks is detected, VM location information as
...
-1. [Retrieve your Tenant ID](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal#get-tenant-id)
+For administrators [creating and managing credentials](https://docs.rightscale.com/policies/users/guides/credential_management.html) to use with this policy, the following information is needed:
 
-## Required RightScale Roles
+Provider tag value to match this policy: `azure_rm`
 
-- credential_viewer or admin
-
-## Azure Required Permissions
+### Azure Required Permissions
```
Comment (cdwilhelm) [inline operational/azure/vms_without_managed_disks/README.md]: Are there permissions needed for the Disk?

File: operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt [hunk 9]
```diff
@@ -137,6 +138,8 @@ script "js_filter_virtual_machines", type: "javascript" do
         }
       }
     })
+    result = _.sortBy(result, 'location');
```
Comment (cdwilhelm) [inline operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt]: sort by subscription name first?

File: operational/azure/vms_without_managed_disks/README.md (hunk 8 above)
Comment (cdwilhelm) [inline operational/azure/vms_without_managed_disks/README.md]: replace with Required permissions in the provider:

File: compliance/azure/ahub_manual/README.md (hunk 1 above)
Comment (KTebrewal) [reply compliance/azure/ahub_manual/README.md]: Updated the missing provider tag and cloud configuration sections

File: compliance/azure/ahub_manual/README.md (hunk 2 above)
Comment (KTebrewal) [reply compliance/azure/ahub_manual/README.md]: There is no such API used that may need specific AHUB permissions.

File: compliance/azure/compliance_score/README.md (hunk 3 above)
Comment (KTebrewal) [reply compliance/azure/compliance_score/README.md]: Updated the missing provider tag and cloud configuration sections

File: compliance/azure/compliance_score/azure_regulatory_compliance_report.pt (hunk 4 above)
Comment (KTebrewal) [reply compliance/azure/compliance_score/azure_regulatory_compliance_report.pt]: Added and Sorted.

File: cost/azure/hybrid_use_benefit/README.md (hunk 5 above)
Comment (KTebrewal) [reply cost/azure/hybrid_use_benefit/README.md]: Updated the missing provider tag and cloud configuration sections.
There is no such API used that may need specific AHUB permissions.

File: cost/azure/hybrid_use_benefit/README.md (hunk 6 above)
Comment (KTebrewal) [reply cost/azure/hybrid_use_benefit/README.md]: Addressed the issue.

File: cost/azure/hybrid_use_benefit/azure_hybrid_use_benefit.pt (hunk 7 above)
Comment (KTebrewal) [reply cost/azure/hybrid_use_benefit/azure_hybrid_use_benefit.pt]: Updated.

File: operational/azure/vms_without_managed_disks/README.md (hunk 8 above)
Comment (KTebrewal) [reply operational/azure/vms_without_managed_disks/README.md]: There is no such API used that may need specific Disk permissions.

File: operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt (hunk 9 above)
Comment (KTebrewal) [reply operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt]: Updated.

File: operational/azure/vms_without_managed_disks/README.md (hunk 8 above)
Comment (KTebrewal) [reply operational/azure/vms_without_managed_disks/README.md]: Updated.

**Pull request merged**